front-end/src/components/modal/upload/UploadModal.jsx

```diff
@@ -50,22 +50,13 @@ export const UploadModal = ({ setModal, setSuccessMessage}) => {
     //start loading
     setLoading(true);
 
-    await new Promise((resolve, reject) => {
+    // TODO: This does not handle uploading multiple, just ensures the last finishes before setting loading to false
+    await new Promise((resolve) => {
       for (let i = 0; i < fileLists.length; i++) {
-        ApiUtil.uploadFile(fileLists[i]).then((res) => {
-          res.text().then(text => {
-            if(res.status !== 200) {
-              alert(JSON.stringify(`${text}, status: ${res.status}`));
-            }
-            if (i===fileLists.length-1) {
-              resolve();
-            }
-          });
-        }).catch(e => { 
-          alert(e);
-          reject(e);
-        });
-        
+        ApiUtil.uploadFile(fileLists[i]);
+        if (i===fileLists.length-1) {
+          resolve();
```

Comment:
How are files that would previously been rejected now handled?

Reply:
ApiUtils now returns the parsed response via 'response.json()' or similar.

Reply:
Used to reject in the promise, now it throws the error application wide. Temporary solution until we improve front end error handling, as all other methods do the same thing.

Reply:
https://github.com/McMaster-Baja-Racing/Better-Data-Viewer/issues/125